godot-core/src/lib.rs

```diff
@@ -77,14 +84,49 @@ pub mod private {
         F: FnOnce() -> R + std::panic::UnwindSafe,
         S: std::fmt::Display,
     {
-        match std::panic::catch_unwind(code) {
+        let info: Arc<Mutex<Option<GodotPanicInfo>>> = Arc::new(Mutex::new(None));
+
+        // Back up previous hook, set new one
+        let prev_hook = std::panic::take_hook();
+        {
+            let info = info.clone();
+            std::panic::set_hook(Box::new(move |panic_info| {
+                if let Some(location) = panic_info.location() {
+                    *info.lock().unwrap() = Some(GodotPanicInfo {
+                        file: location.file().to_string(),
+                        line: location.line(),
+                    });
+                } else {
+                    println!("panic occurred but can't get location information...");
+                }
+            }));
+        }
+
+        // Run code that should panic, restore hook
+        let panic = std::panic::catch_unwind(code);
+        std::panic::set_hook(prev_hook);
+
+        match panic {
             Ok(result) => Some(result),
             Err(err) => {
                 // Flush, to make sure previous Rust output (e.g. test announcement, or debug prints during app) have been printed
                 // TODO write custom panic handler and move this there, before panic backtrace printing
                 flush_stdout();
 
-                log::godot_error!("Rust function panicked. Context: {}", error_context());
+                log::godot_error!(
+                    "Rust function panicked in file {} at line {}. Context: {}",
+                    info.lock()
+                        .unwrap()
+                        .as_ref()
+                        .expect("No file provided")
+                        .file,
+                    info.lock()
+                        .unwrap()
+                        .as_ref()
+                        .expect("No line provided")
+                        .line,
+                    error_context()
+                );
```

Comment:
Now that you have one struct, you need to check presence only once:

```suggestion
                let info = info.lock.unwrap().as_ref().expect("no panic info available");
                log::godot_error!(
                    "Rust function panicked in file {} at line {}. Context: {}",
                    info.file, info.line, error_context()
                );
```